lib/components/molecules/DateSelector/DateSelector.js

```diff
@@ -0,0 +1,184 @@
+  };
+
+  const getMonthOptions = () => {
+    let monthOptions = [...months];
+    if (selectedDate.getFullYear() === startDate.getFullYear()) {
+      const startMonth = startDate.getMonth();
+      monthOptions = monthOptions.slice(startMonth, months.length);
+    }
+    if (selectedDate.getFullYear() === endDate.getFullYear()) {
+      const endMonth = endDate.getMonth();
+      monthOptions = monthOptions.slice(0, endMonth + 1);
+    }
+    return monthOptions;
+  };
+
+  const getYearOptions = () => {
+    const years = [];
+    for (let i = startDate.getFullYear(); i <= endDate.getFullYear(); i += 1) {
+      years.push(i);
+    }
+    return years;
+  };
+
+  const DaySelector = (
+    <>
+      <Label htmlFor={`${id}-day`}>{dateLabel}</Label>
+      <Select
+        onChange={e => {
+          return setSelectedDate(
+            new Date(selectedDate.getFullYear(), selectedDate.getMonth(), e.target.value)
+          );
+        }}
+        id={`${id}-day`}
+        name="day"
+        className={classNames('date', className)}
+        selectedOption={selectedDate.getDate()}
+        options={getDaysInMonth()}
+      />
+    </>
+  );
+
+  const MonthSelector = (
+    <>
+      <Label htmlFor={`${id}-month`}>{monthLabel}</Label>
+      <Select
+        onChange={e => {
+          return setSelectedDate(
+            new Date(
+              selectedDate.getFullYear(),
+              months.indexOf(e.target.value),
+              selectedDate.getDate()
+            )
+          );
+        }}
+        id={`${id}-month`}
+        name="month"
+        className={classNames('month', className)}
+        selectedOption={months[selectedDate.getMonth()]}
+        options={getMonthOptions()}
+      />
+    </>
+  );
+
+  const YearSelector = (
+    <>
+      <Label htmlFor={`${id}-year`}>{yearLabel}</Label>
+      <Select
+        onChange={e => {
+          return setSelectedDate(
+            new Date(e.target.value, selectedDate.getMonth(), selectedDate.getDate())
+          );
+        }}
+        id={`${id}-year`}
+        name="year"
+        className={classNames('year', className)}
+        selectedOption={selectedDate.getFullYear()}
+        options={getYearOptions()}
+      />
+    </>
+  );
+
+  const renderSelector = value => {
+    switch (value.toLowerCase()) {
+      case 'mmddyy':
+      case 'mmddyyyy':
+        return [MonthSelector, DaySelector, YearSelector];
+      case 'mmyy':
+      case 'mmyyyy':
+        return [MonthSelector, YearSelector];
+      case 'ddmmyy':
+      case 'ddmmyyyy':
+      default:
+        return [DaySelector, MonthSelector, YearSelector];
+    }
+  };
+
+  return <div className={className}>{renderSelector(format)}</div>;
+};
+
+DateSelector.defaultProps = {
+  id: 'date-selector',
+  className: 'date-selector',
+  format: 'ddmmyy',
+  startDate: new Date(),
+  endDate: new Date(new Date().setFullYear(new Date().getFullYear() + 50)),
+  locale: 'default',
+  dateLabel: 'Date',
```

Comment:
Don't set below as default value, Make them mandatory, Remove the showLabel: false. For accessibility its important that every field has a value.  

1. dateLabel: 'Date',
2. monthLabel: 'Month',
3. yearLabel: 'Year',